Initial postgresport setup

sravani1510 opened the pull request:
Includes patch files up to the build stage.

Commit:
Initial postgresport commit

HarithaIBM commented on patches/Makefile_shlib_patch.patch:
```diff
@@ -0,0 +1,46 @@
++  LINK.shared   	= $(COMPILER)
++  ifdef SHLIB_EXPORTS
++    exports_file	= $(SHLIB_EXPORTS:%.txt=%.x)
++#    $(exports_file): $(SHLIB_EXPORTS)
```
can we remove these commented lines

sravani1510 replied:
Yes removed, addressed in Point-1 in commit aff67132c5c61fe3cdabd07143baba33aaf8d543

HarithaIBM commented on patches/data_patch.patch:
```diff
@@ -0,0 +1,22 @@
+ 	 */
+ 	text_to_cstring_buffer(zone, tzname, sizeof(tzname));
+ 
+-	type = DecodeTimezoneName(tzname, &val, &tzp);
```
can we guard it with #ifdef __MVS__ in such cases

sravani1510 replied:
Addressed this change in Point-2 in commit aff67132c5c61fe3cdabd07143baba33aaf8d543

Commit:
Modified timestamp patch to match the current version code

VikasPatnala committed:
These patches contains changes for configure step for zos

VikasPatnala committed:
Adding postgres TAG REL_18_BETA1 to buildenv

VikasPatnala committed:
adding missing part in the patch

VikasPatnala committed:
[runtime error] patch addressing runtime error - From entry point pg_tzset at compile unit offset +64206E6F7420696E at entry offset +64206E6F4BD209

VikasPatnala committed:
[runtime error] patch addressing runtime error - could not map anonymous shared memory: EDC5113I Bad file descriptor.

VikasPatnala committed:
[runtime error] patch addressing runtime error - From entry point pg_get_encoding_from_locale at compile unit

IgorTodorovskiIBM commented on buildenv:
```diff
@@ -0,0 +1,184 @@
+## Specify the default build line, either 'DEV' or 'STABLE' (default: STABLE)
+export ZOPEN_BUILD_LINE="DEV"
+
+## Required IF ZOPEN_BUILD_LINE='STABLE'
```
Should we target a STABLE release? Maybe https://github.com/postgres/postgres/releases/tag/REL_18_BETA2 ?
You can set export ZOPEN_STABLE_TAG=REL_18_BETA2

sravani1510 replied:
Addressed this change in commit aff67132c5c61fe3cdabd07143baba33aaf8d543

IgorTodorovskiIBM commented on buildenv:
```diff
@@ -0,0 +1,184 @@
+export ZOPEN_CATEGORIES="database"
+
+## Runtime zopen tool dependencies to be installed alongside the tool.
+export ZOPEN_RUNTIME_DEPS="zlib libxml2 libxslt lz4 systemd tcl python perl"
```
If they're statically linked, then we don't need to add the libraries are runtime dependencies

IgorTodorovskiIBM commented on patches/fd_c_patch.patch:
```diff
@@ -0,0 +1,14 @@
+ 
+ retry:
++	#ifndef __MVS__
+ 	rc = fdatasync(fd);
```
Can we replace this with fsync?

sravani1510 replied:
yeah, replaced with fsync and addressed this change in commit aff67132c5c61fe3cdabd07143baba33aaf8d543

IgorTodorovskiIBM commented on patches/findtimezone_patch.patch:
```diff
@@ -0,0 +1,195 @@
+@@ -231,7 +231,7 @@ compare_tm(struct tm *s, struct pg_tm *p)
+  * test time.
+  */
+ static int
```
Would a `#define tzname _tzname` be a less intrusive way to fix this?

sravani1510 replied:
Addressed this change in Point 2 in commit aff67132c5c61fe3cdabd07143baba33aaf8d543

IgorTodorovskiIBM commented on patches/pg_test_fsync_c_patch.patch:
```diff
@@ -0,0 +1,14 @@
+ 						  writes * XLOG_BLCKSZ) != XLOG_BLCKSZ)
+ 				die("write failed");
++#ifndef __MVS__
+ 		fdatasync(tmpfile);
```
Suggest changing to fsync for MVS

sravani1510 replied:
addressed in point 3 in commit aff67132c5c61fe3cdabd07143baba33aaf8d543

IgorTodorovskiIBM commented on patches/waiteventset_patch.patch:
```diff
@@ -0,0 +1,139 @@
+diff --git a/src/backend/storage/ipc/waiteventset.c b/src/backend/storage/ipc/waiteventset.c
+index 7c0e66900f9..574ced67132 100644
+--- a/src/backend/storage/ipc/waiteventset.c
```
I'd suggest unsetting WAIT_USE_EPOLL in case you miss a few conditions now or in the future.

`#unset WAIT_USE_EPOLL`

Commit:
remove extra warnings by binder

sravani1510 committed:
Addressing Review Comments
1. Removed unneccessary comments in file src/Makefile.shlib and added few comments for readablity
2. Added #define for tzname conflict
	#ifdef __MVS__
	#define tzname _tzname
	#endif
3. Added fsync for __MVS__ target instead of fdatasync

VikasPatnala committed:
Update README.md till current changes

VikasPatnala committed:
Update README.md
